definitions/scenarios/self_upgrade.rb

```diff
@@ -0,0 +1,90 @@
+module ForemanMaintain::Scenarios
+  class SelfUpgradeBase < ForemanMaintain::Scenario
+    def enabled_system_repos_id
+      feature(:system_repos).enabled_repos_ids
+    end
+
+    def enable_repos(repo_ids = stored_enabled_repos_ids)
+      add_step(Procedures::Repositories::Enable.new(repos: repo_ids))
+    end
+
+    def disable_repos(repo_ids = stored_enabled_repos_ids)
+      add_step(Procedures::Repositories::Disable.new(repos: repo_ids))
+    end
+
+    def target_version
+      @target_version ||= context.get(:target_version)
+    end
+
+    def current_version
+      feature(:instance).downstream.current_minor_version
+    end
+
+    def maintenance_repo(version)
+      @maintenance_repo ||= "rhel-#{el_major_version}-server-satellite-maintenance-#{version}-rpms"
+    end
```

Comment:
This is an odd caching because it has a parameter. Consider this:
```ruby
self_upgrade_base.maintenance_repo('6.9')
self_upgrade_base.maintenance_repo('6.10')
```

The first invocation will return 6.9 (as you'd expect, but the second will also return 6.9. That's surprising. It looks like this isn't used elsewhere either so I'd drop it.
```suggestion
    def maintenance_repo(version)
      "rhel-#{el_major_version}-server-satellite-maintenance-#{version}-rpms"
    end
```